deploy/helm/kubecf/assets/operations/instance_groups/bits.yaml

```diff
@@ -12,8 +12,8 @@
   path: /releases/name=bits-service/stemcell?
```

Comment:
I think the whole operation should be conditional here and below. The purpose of the `releases` key on `values.yaml` is to allow passing specifics per release. https://github.com/SUSE/kubecf/blob/master/deploy/helm/kubecf/values.yaml#L15
E.g.

```yaml
releases:
  bits-service:
    stemcell:
      os: opensuse-42.3
      version: 36.g03b4653-30.80-7.0.0_340.g2b599a90
```

So, while we don't have a more generic mechanism for setting those, we can conditionally render this `replace` if `releases.bits-service.stemcell` is set.

```yaml
{{- if .Values.relases.bits-service.stemcell }}
- type: replace
  path: /releases/name=bits-service/stemcell?
  value: {{ .Values.releases.bits-service.stemcell | mustToJson }}
{{- end }}
```
( I didn't test my snippet above - typed directly in the browser)

Reply:
This is _pure bikeshedding_ and you definitely don't have to even think about this, but we can alternatively do (also untested):
```yaml
- type: replace
  path: /releases/name=bits-service/stemcell?
  value: {{ default .Values.releases.defaults.stemcell (index (default .Values.releases.bits-service (dict)) "stemcell") | mustToJson }}
```
And possibly do this with a template helper or something.

… it seemed more sane before I typed it all out, really.

Reply:
Thanks for your feedback! I really appreciate the input. At first, I tried the one-liner as @mook-as suggested, but it quickly became really ugly to read. In my opinion, Go Template is not really pleasant to read in the first place, and the my one-liner attempt even made it harder to read. This is why I opted for a helper function like Mark suggested, too. Let me know what you think about it. I do not have a ton of experience with Go templates, so please tell me if this is not the way to do it or whether there would be a more elegant way.

Reply:
Perfect.